core/src/main/java/tc/oc/pgm/api/channels/Channel.java

```diff
@@ -0,0 +1,118 @@
+import cloud.commandframework.paper.PaperCommandManager;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Predicate;
+import net.kyori.adventure.text.Component;
+import net.kyori.adventure.text.format.NamedTextColor;
+import org.bukkit.command.CommandSender;
+import org.jetbrains.annotations.Nullable;
+import tc.oc.pgm.api.PGM;
+import tc.oc.pgm.api.event.ChannelMessageEvent;
+import tc.oc.pgm.api.player.MatchPlayer;
+import tc.oc.pgm.api.setting.SettingValue;
+import tc.oc.pgm.util.Players;
+import tc.oc.pgm.util.named.NameStyle;
+
+public interface Channel<T> {
+
+  default String getDisplayName() {
+    return getAliases()[0];
+  }
+
+  String[] getAliases();
+
+  @Nullable
+  default Character getShortcut() {
+    return null;
+  }
+
+  default SettingValue getSetting() {
+    return null;
+  }
+
+  default boolean supportsRedirect() {
+    return false;
+  }
+
+  default boolean canSendMessage(MatchPlayer sender) {
+    return true;
+  }
+
+  T getTarget(MatchPlayer sender, Map<String, ?> arguments);
+
+  Collection<MatchPlayer> getViewers(T target);
+
+  default Collection<MatchPlayer> getBroadcastViewers(T target) {
+    return getViewers(target);
+  }
+
+  default void sendMessage(ChannelMessageEvent<T> event) {}
+
+  default Component formatMessage(T target, @Nullable MatchPlayer sender, Component message) {
+    if (sender == null) return message;
+    return text()
+        .append(text("<", NamedTextColor.WHITE))
+        .append(sender.getName(NameStyle.VERBOSE))
+        .append(text(">: ", NamedTextColor.WHITE))
+        .append(message)
+        .build();
+  }
+
+  default void registerCommand(PaperCommandManager<CommandSender> manager) {
+    String name = getAliases()[0];
+    String[] aliases = new String[getAliases().length - 1];
+    System.arraycopy(getAliases(), 1, aliases, 0, aliases.length);
+
+    manager.command(
+        manager
+            .commandBuilder(name, aliases)
+            .handler(
+                context -> {
+                  MatchPlayer sender =
+                      context.inject(MatchPlayer.class).orElseThrow(IllegalStateException::new);
+                  PGM.get().getChannelManager().setChannel(sender, this);
+                }));
+
+    manager.command(
+        manager
+            .commandBuilder(name, aliases)
+            .argument(
+                StringArgument.<CommandSender>builder("message")
+                    .greedy()
+                    .withSuggestionsProvider(Players::suggestPlayers)
+                    .build())
+            .handler(
+                context -> {
+                  MatchPlayer sender =
+                      context.inject(MatchPlayer.class).orElseThrow(IllegalStateException::new);
+                  PGM.get().getChannelManager().process(this, sender, context.asMap());
+                }));
+  }
+
+  default Map<String, Object> processChatShortcut(MatchPlayer sender, String message) {
```

Comment:
```suggestion
  default Map<String, Object> processShortcutMessage(MatchPlayer sender, String message) {
```
you're not processing the shortcut (with a chat), you're processing the message (with a shortcut).

Make sure the docs explain the rationale of why this is even needed cause it's nonobvious